iocore/net/quic/QUICIncomingFrameBuffer.cc

```diff
@@ -78,30 +80,30 @@ QUICIncomingStreamFrameBuffer::pop()
 QUICConnectionErrorUPtr
 QUICIncomingStreamFrameBuffer::insert(const QUICFrame &frame)
 {
-  const QUICStreamFrame *stream_frame = static_cast<const QUICStreamFrame *>(&frame);
+  const QUICStreamFrame &stream_frame = static_cast<const QUICStreamFrame &>(frame);
 
-  QUICOffset offset = stream_frame->offset();
-  size_t len        = stream_frame->data_length();
+  QUICOffset offset = stream_frame.offset();
+  size_t len        = stream_frame.data_length();
 
-  QUICConnectionErrorUPtr err = this->_check_and_set_fin_flag(offset, len, stream_frame->has_fin_flag());
+  QUICConnectionErrorUPtr err = this->_check_and_set_fin_flag(offset, len, stream_frame.has_fin_flag());
   if (err != nullptr) {
     return err;
   }
 
   // Ignore empty stream frame except pure fin stream frame
-  if (len == 0 && !stream_frame->has_fin_flag()) {
+  if (len == 0 && !stream_frame.has_fin_flag()) {
     return nullptr;
   }
 
   if (this->_recv_offset > offset) {
     // dup frame;
     return nullptr;
   } else if (this->_recv_offset == offset) {
-    this->_recv_offset   = offset + len;
-    QUICFrameSPtr cloned = frame.clone();
+    this->_recv_offset = offset + len;
+    QUICFrame *cloned  = new QUICStreamFrame(stream_frame);
```

Comment:
We need to create a copy because the original frame is temporal, the actual data (payload) won't be copied though. Any ideas?

Reply:
Creating a instance before passing it `insert` may things simpler a bit?

Reply:
>Yes we should delete that. Since we only need `offset` and `IOBufferBlock`.